### DISCUSSION TITLE
Adding a wait time between submissions

### DISCUSSION THREAD

**Discussion by raypettit57** [gradescope/autograder_samples#59] [Ideas]
I occasionally notice students who attempt to improve their program by making almost random changes and resubmitting, hoping to get lucky. I also see students who will ask a TA for help, not understand the advice, but type it in and resubmit immediately, and try to 'kidnap' the TA until all tests are passed. While both of these examples involve poor decisions on the part of the student, I would like to be able to add a wait time in the system to eliminate this behavior. And I guess the wait time could be in receiving feedback or in resubmitting, but having the option for both would also be nice.

**Comment by DaveedaKinG** [comment]
Hey you’re right, but we can’t judge students, cause they are in their leaning process and more so the level of assimilation 
Is very likely not to same

**Comment by ibrahima** [comment]
Hi @raypettit57! Thanks for the suggestion. That is a great idea and it is a feature request that we have gotten in the past. While it's not something that's built into the Gradescope app directly, you can implement some type of rate limiting scheme within your autograder by looking at the submission times in the [submission metadata file](https://gradescope-autograders.readthedocs.io/en/latest/submission_metadata/#rate-limiting-schemes). It would be great to have it in the app so that students can be prevented from submitting in a more user-friendly manner. Though one benefit of implementing it in the autograder is that it gives maximum flexibility to the instructors to design a custom rate limiting policy.

For example, one of our users (now an engineer on the team!) built a [rate limiting capability](https://github.com/ThaumicMekanism/GradescopeBase/blob/master/GradescopeBase/AutograderRateLimit.py) in his Python autograder framework. You can see [an example of how to use his framework here](https://github.com/ThaumicMekanism/GradescopeBaseAG/blob/master/main.py#L22).
